### PR TITLE
Ensure CVE-2015-9284 is actually mitigated

### PR DESCRIPTION
**Pull request by amichal**

### CODE REVIEW

File: .env.test [hunk 1]
```diff
@@ -1,9 +1,4 @@
 RAILS_ENV=test
-DATABASE_APP_DB_TEST=warehouse_app_test
-WAREHOUSE_DATABASE_DB_TEST=warehouse_test
-HEALTH_DATABASE_DB_TEST=health_test
-REPORTING_DATABASE_DB_TEST=reporting_test
-WAREHOUSE_DATABASE_ADAPTER=postgis
 FQDN=hmis-warehouse.test
```
Comment (eanders) [inline .env.test]: Why are we removing these?

Comment (amichal) [reply .env.test]: I had a `.env` file locally with identical settings and no secrets. I use `.env.local` most of the time which overrode all of them so I thought this project had a `.env` committed with sensible defaults and had forgotten that is/was part of the initial setup scripts.

**Comment by amichal** [comment]
I need to figure out how we want to enable OKTA/omni for that test only.. The failing tests are because with Omniauth enabled we change the login screens

**Comment by eanders** [comment]
Can you inject the ENV just for that one set of test, then set them to nil afterward?

**Comment by amichal** [comment]
Since its an integration test the rack server appears to be started once in the runner and so the ENV changes come too late to change the behaviour (all the important ENV checks happen at process load time). I grepped around thinking I could maybe trigger a new server with `IntegrationTest#open_session` (which ive used in minitest) but was unsure I could make that work or figure out its rspec variant. Spent some time trying to learn rspec "tags" etc and then punted and just put the entire test is a if ENV.. present? and ran rspec twice in the runner.

**Comment by amichal** [comment]
In some future version of this feature maybe we move the omniauth support to a runtime config, have the routes always present and toggle the functionality in and out via the database or whatever. Or maybe I'll get better at rspec and find a elegant way to run some tests under a different rack process

**Pull request merged**